src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -952,14 +951,30 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
                         for (i = 0; (uint)i < maxIters; i++)
                         {
                             utf16Data = Unsafe.ReadUnaligned<Vector128<short>>(pInputBuffer);
-                            if (!Sse41.TestZ(utf16Data, nonAsciiUtf16DataMask))
+
+                            if (AdvSimd.IsSupported)
                             {
-                                goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
-                            }
+                                Vector128<short> isUtf16DataNonAscii = AdvSimd.CompareTest(utf16Data, nonAsciiUtf16DataMask);
+                                bool hasNonAsciiDataInVector = AdvSimd.Arm64.MaxPairwise(isUtf16DataNonAscii, isUtf16DataNonAscii).AsUInt64().ToScalar() != 0;
```

Comment:
I think this line is wrong and probably causes the tests failures. 

Suppose you have `isUtf16DataNonAscii[0]` equal to `0xFFFF`  (i.e. `-1`) and `isUtf16DataNonAscii[1]` equal to `0`. Then `MaxPairwise(isUtf16DataNonAscii, isUtf16DataNonAscii)` will return `0` in the 0-the lane of the resulting value.

You need to use `MinPairwise` here. You would use `MaxPairwise` if the element type was unsigned.

This is why I originally suggested to use `MinAcross` for this.

In my reply to Kunal I also assumed the unsigned type of the values (see https://github.com/dotnet/runtime/pull/39041#discussion_r455225992)

Reply:
Good catch @echesakovMSFT . The line above should be `Vector128<ushort> isUtf16DataNonAscii = AdvSimd.CompareTest(utf16Data, nonAsciiUtf16DataMask).AsUInt16()`.

Reply:
Thanks for helping find the root cause. I'm working on this now.